presto-main/src/main/java/com/facebook/presto/sql/relational/DeterminismEvaluator.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.relational;
+
+public class DeterminismEvaluator
+{
+    private DeterminismEvaluator() {}
+
+    public static boolean isDeterministic(RowExpression expression)
+    {
+        return expression.accept(new Visitor(), null);
+    }
+
+    private static class Visitor
+            implements RowExpressionVisitor<Void, Boolean>
+    {
+        @Override
+        public Boolean visitInputReference(InputReferenceExpression reference, Void context)
+        {
+            return true;
+        }
+
+        @Override
+        public Boolean visitConstant(ConstantExpression literal, Void context)
+        {
+            return true;
+        }
+
+        @Override
+        public Boolean visitCall(CallExpression call, Void context)
+        {
+            // TODO: total hack to figure out if a function is deterministic. martint should fix this when he refactors the planning code
+            if (call.getSignature().getName().equals("random") || call.getSignature().getName().equals("rand")) {
```

Comment:
@erichwang or @martint is there is better way to know if a function is deterministic?  Can we look up in the function registry?